internal/controller/processor.go

```diff
@@ -201,6 +210,7 @@ func (processor *AppConfigurationProviderProcessor) processSecretReferenceRefres
 
 	// Only resolve the secret references that not specified the secret version
 	secretReferencesToSolve := make(map[string]*loader.TargetSecretReference)
+	cachedSecretReferences := make(map[string]*loader.TargetSecretReference)
```

Comment:
This variable seems useless, do we really need it?

Reply:
Currently  `ResolveSecretReferences` doesn't pass/return new parameters for `secretID` of latest secret, it just update the `secretID` in `kvReferencesToResolve` . So need to cache secretRef before resolving.